### PR TITLE
Roi component

### PR DESCRIPTION
**Pull request by gamaanderson**
This is a suggestion for a change in ROI and I want to emphasize as a suggestion this can also be rejected, I can also see reasons for that. This is mostly personal preference and a sense of what is better that I can't really explain, there is however one positive point, the new ROI is independent of Vradar and therefore can be used (in the future) with grid.

The new ROI may be found in the components menu in the menubar. But this is open to discussion.

Take your time with this, this is NOT a priority and I pretend to focus more in the documentation

### CODE REVIEW

**Comment by nguy** [comment]
I like the idea behind this and was thinking of something similar in terms of relocating the ROI function as a component. I'll need to look through and see how this is set up. 

By the way, is there documentation of how to call a component? I think this will be important to let users know how to use the plugins effectively.

**Comment by gamaanderson** [comment]
I'm working in a tutorial for writing scripts right now, this may help, but keep this idea we will need it

**Comment by nguy** [comment]
Good deal, thanks!

I've been doing some mental coding (trying to get a structure in my head) with some of these issues. I am trying to decide what would be really nice to have done by Sep AMS radar conference. We'll be presenting ARTView as an option in the open source software short course. So any tutorials you think might be useful please let me know. 

Hopefully I can get back to this soon. I have a couple other things I need to finish up.

**Comment by gamaanderson** [comment]
You can keep up with what I'm doing [here](https://rawgit.com/gamaanderson/artview/documentation/docs/build/html/index.html). The reference manual is a lot more complete, the tutorial is a work in progress. Just let me say I'm famous for some terrible spelling errors, there is probably a lot there.

If you have clear ideas about what you need, let me know.

File: artview/components/plot_radar.py [hunk 1]
```diff
@@ -466,6 +466,30 @@ def toolDefaultCmd(self):
...
+        '''
+        from .tools import interior
+        radar = self.Vradar.value
+        xy, idx = interior(path, self.Vradar.value, self.Vtilt.value)
+        aux = (xy[:, 0], xy[:, 1], radar.azimuth['data'][idx[:, 0]],
+               radar.range['data'][idx[:, 1]] / 1000.,
+               radar.fields[self.Vfield.value]['data'][idx[:, 0], idx[:, 1]],
+               idx[:, 0], idx[:, 1])
+        return aux
+
```
Comment (nguy) [inline artview/components/plot_radar.py]: I think this is a really good idea. This routine should become useful with future development.

File: artview/components/plot_radar.py [hunk 2]
```diff
@@ -657,6 +681,16 @@ def _savefile(self, PTYPE=IMAGE_EXT):
...
+    ########################
+    #      get methods     #
+    ########################
+
+    def getPlotAxis(self):
+        return self.ax
+
+    def getStatusBar(self):
+        return self.statusbar
+
```
Comment (nguy) [inline artview/components/plot_radar.py]: Nice, these are much more elegant solutions!

File: artview/components/plot_radar.py (hunk 1 above)
Comment (gamaanderson) [inline artview/components/plot_radar.py]: The base here is that ROI is about space, not the data that is in that space, so it should not have access to Vradar. As a consequence Display must provide this information.

**Comment by nguy** [comment]
Okay, so this all seems to work just fine. 
I guess I don't really know how the components work. I see that ROI is now a component as well as tool (let's keep it that way for now). But selecting it does not "run" anything.

Could you note the procedure here or elsewhere (maybe you've done this already and I don't recall)?

**Comment by gamaanderson** [comment]
Ok, that is missing a warning, you must select a display before starting it. You probably just clicked start right way and that resulted in an error in the terminal

**Pull request merged**